src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java

```diff
@@ -0,0 +1,240 @@
+    }
+
+    /**
+     * Will return the Worker interface which should be used to identify workers for the given platform
+     *
+     * @param name The name of the platform
+     * @return The interface used to identify a worker
+     */
+    public Worker getWorker(String name) {
+        return getPlatform(name)
+                .orElseThrow(() -> new IllegalArgumentException("Platform not found"))
+                .getWorker().orElse(fallbackWorker);
+    }
+
+    /**
+     * Will return the payment service which should be used for the given platform
+     * If there is no Platform with the given name None is returned.
+     *
+     * @param name The name of the platform to use
+     * @return The interface used for payment
+     */
+    public Payment getPlatformPayment(String name) {
+        return getPlatform(name)
+                .orElseThrow(() -> new IllegalArgumentException("Platform not found"))
+                .getPayment().orElse(fallbackPayment);
+    }
+
+    /**
+     * Publish the given experiment on the platform.
+     * The method will update the database with the new public task
+     *
+     * @param name The name of the platform
+     * @param experiment The experiment to publish
+     * @return None if the platform does not exist
+     */
+    public CompletableFuture<Boolean> publishTask(String name, Experiment experiment) throws TaskOperationException {
+
+        TaskRecord record = new TaskRecord();
+        record.setExperiment(experiment.getId());
+        record.setStatus(TaskStatus.running);
+        record.setCrowdPlatform(name);
+
+        TaskRecord result = tasksOps.createTask(record);
+        if (result == null)
+            throw new TaskOperationException("Task could not be created");
+
+        return getPlatform(name)
+                .map(platform1 -> platform1.publishTask(experiment))
+                .orElseThrow(() -> new IllegalArgumentException("Platform not found!"))
+                .handle((s1, throwable) -> {
+                    if (s1 != null) {
+                        result.setPlatformData(s1);
+                    } else {
+                        result.setStatus(TaskStatus.stopped);
+                    }
+                    if (!tasksOps.updateTask(result)) {
+                        throw new IllegalStateException("Updating record for published task failed");
+                    }
+                    return true;
+                });
+    }
+
+    /**
+     * Unpublish a given experiment from the given platform
+     *
+     * @param name The name of the platform
+     * @param experiment The experiment to unpublish
+     * @return None if the platform was not found, false if the unpublish failed and true if everything went fine
+     */
+    public CompletableFuture<Boolean> unpublishTask(String name, Experiment experiment) throws TaskOperationException {
+        TaskRecord record;
+
+        record = tasksOps.searchTask(name, experiment.getId()).orElse(null);
```

Comment:
i would first set it to unpublished, so no other thread can unpublish it at the same time